content/200-concepts/100-components/02-prisma-client/250-tracing.mdx

```diff
@@ -0,0 +1,336 @@
+
+</Admonition>
+
+</TopBlock>
+
+## About tracing
+
+When you enable tracing, Prisma Client outputs the following:
+
+- One trace for each operation that Prisma Client makes.
+- In each trace, one or more [spans](https://www.opentelemetry.io/docs/reference/specification/trace/api/#span). Each span represents the length of time that one stage of the operation takes, such as serialization, or a database query.
+
+The number and type of spans in a trace varies depending on the type of operation the trace covers, but an example is as follows:
+
+<!--//diagram source file: https://www.figma.com/file/iUpH3SKOBFUrz6rh0LKP2q/Andrew's-diagrams-(temp)?node-id=0%3A1
+![image](tracing-diagram.png)-->
+
+You can send tracing output to the console, or analyze it in any OpenTelemetry-compatible tracing system, such as [Jaeger](https://www.jaegertracing.io/), [Honeycomb](https://www.honeycomb.io/trace/) and [Datadog](https://www.datadoghq.com/). On this page, we give an example of how to send tracing output to Jaeger, which you can [run locally](#run-jaeger-locally).
+
+## Prisma's Spans
+
+For each trace, Prisma Client outputs a series of spans. The number and type of these spans varies, depending on the Prisma operation. A typical Prisma trace has the following spans:
+
+- `prisma`: Represents the entire Prisma operation, from Prisma Client to the database and back. It contains details such as the model and method called by Prisma Client. Depending on the Prisme operation, it contains one or more of the following spans:
+  - `prisma:query_builder`: Represents how long a query takes in the [query engine](/concepts/components/prisma-engines/query-engine).
+    - `prisma:connection`: Represents how long it took for Prisma Client to get a database connection.
+    - `prisma:db_query`: Represents a SQL query that was executed against the database. It includes the SQL statement or command (in the tags), and how long the query took to run.
+
+For example, given the following Prisma code:
+
+```ts
+prisma.user.findMany({
+  where: {
+    email: email,
+  },
+  include: {
+    posts: true,
+  },
+})
+```
+
+The trace is structured as follows:
+
+- `prisma`
+  - `prisma:query_builder`:
+    - `prisma:connection`
+    - `prisma:db_query`: details of the first SQL query or command...
+    - `prisma:db_query`: ...details of the next SQL query or command...
+    - `prisma:db_query`: ...and so on, for each SQL query or command
+
+### Middleware
+
+By default, tracing does not output spans for middleware. To include your [middleware](/concepts/components/prisma-client/middleware) in your traces, set `middleware` to `true` in your `registerInstrumentations` statement:
+
+```ts
+registerInstrumentations({
+  instrumentations: [new PrismaInstrumentation({ middleware: true })],
+})
+```
+
+This will add an additional type of span to your traces:
+
+- `prisma:middleware`: Represents how long the operation spent in your [middleware](/concepts/components/prisma-client/middleware).
```

Comment:
Agreed. I just moved it to the Going Further section. I'm still not entirely satisfied because it's a mix between overview information of the feature but it depends on enabling tracing. I'll defer to @andrew-walford-prisma to decide how he wants to structure this when he's back :-)